### PR TITLE
Refactor tests to use testify require library

### PR DESCRIPTION
**Pull request by miketonks-form3**
Refactor test to make code cleaner and easier to read, as baseline for future improvements.  The testify library provides excellent helper functions for test assertions, let's use them.

Work towards of investment time project: https://github.com/form3tech/investment-time-project/issues/62

Co-authored-by: Andrea Rosa <andrea.rosa@form3.tech>
Co-authored-by: Kevin Intriago <kevin.intriago@form3.tech>

### CODE REVIEW

**Review by georgijd-form3 — APPROVED** [review]

**Comment by henrique-pinto-form3** [comment]
Hey. I came here because I really dislike assertion libraries and was curious about this pull request. The reasons I dislike are here: https://go.dev/doc/faq#assertions, but this isn't exactly why I'm writing this comment.

The thing that concerned me is that I see that testify/require was used, and it calls t.FailNow (as in, t.Fatal).
Using t.Fatal instead of t.Error on concurrent code messes up the error propagation, and I guess probably wants to be avoided in testing code for testing something concurrent: https://go.dev/doc/go1.16#vet-testing-T

> The vet tool now warns about invalid calls to the testing.T method Fatal from within a goroutine created during the test. This also warns on calls to Fatalf, FailNow, and Skip{,f,Now} methods on testing.T tests or testing.B benchmarks.
> 
> Calls to these methods stop the execution of the created goroutine and not the Test* or Benchmark* function. So these are [required](https://go.dev/pkg/testing/#T.FailNow) to be called by the goroutine running the test or benchmark function.

If you really want to use an assertion library, you might want to consider using another package (like, ../assert) which uses t.Fail instead (as in, t.Error) to make sure you can use your checks inside goroutines concurrently safely.

**Comment by miketonks-form3** [comment]
Thanks @henrique-pinto-form3 for the feedback.  I wasn't aware of this issue and it is always good to learn something new.

It's interesting, perhaps slightly worrying,  that `go vet` can't spot the issue.

I've updated the PR to use `assert` instead which on reflection is closer to the original behavior.

**Comment by henrique-pinto-form3** [comment]
Using FailNow sometimes make sense. Not as common, but not a defect, so it wouldn't make sense for go vet to report it. Static analyzers can only do so much!

File: internal/app/concurrent_proxy_stage_test.go [hunk 1]
```diff
@@ -144,42 +147,28 @@ func (s *ConcurrentProxyStage) the_concurrent_requests_are_sent() {
 		return nil
 	})
 
-	if err != nil {
-		s.t.Error(err)
-	}
+	s.require.NoError(err)
```
Comment (ryangough-f3) [inline internal/app/concurrent_proxy_stage_test.go:150]: There was a general comment about calling `FailNow` from @henrique-pinto-form3 . Doesn't this method still potentially end up calling `t.FailNow` under the hood?

File: internal/app/proxy_stage_test.go [hunk 2]
```diff
@@ -356,126 +351,92 @@ func (s *ProxyStage) multiple_requests_are_sent(requestsToSend int32) {
...
+				// s.t.Fatal("oops") // This exits the inner func instead of TestFoo.
+				// fatal(s)
 			}
 		}()
 
-		if err := s.proxy.WaitForInteraction(s.pactName, int(requestsToSend)); err != nil {
-			s.t.Error(err)
-			s.t.Fail()
-		}
+		go func() {
```
Comment (ryangough-f3) [inline internal/app/proxy_stage_test.go:365]: :confused: not sure what is going on here?

File: internal/app/proxy_stage_test.go [hunk 3]
```diff
@@ -356,126 +351,92 @@ func (s *ProxyStage) multiple_requests_are_sent(requestsToSend int32) {
...
 
 				req.Header.Set("Content-Type", "application/json")
 				atomic.AddInt32(&s.requestsSent, 1)
-				if _, err = http.DefaultClient.Do(req); err != nil {
-					s.t.Error(err)
-					s.t.Fail()
-				}
+				_, err = http.DefaultClient.Do(req)
+				s.assert.NoError(err)
+				// s.t.Fatal("oops") // This exits the inner func instead of TestFoo.
```
Comment (ryangough-f3) [inline internal/app/proxy_stage_test.go:360]: commented code

File: internal/app/concurrent_proxy_stage_test.go (hunk 1 above)
Comment (sata-form3) [reply internal/app/concurrent_proxy_stage_test.go:150]: I don't think this is a problem, we're not creating a goroutine as part of the stage setup?

File: internal/app/proxy_stage_test.go (hunk 3 above)
Comment (miketonks-form3) [reply internal/app/proxy_stage_test.go]: oops

File: internal/app/proxy_stage_test.go (hunk 2 above)
Comment (miketonks-form3) [reply internal/app/proxy_stage_test.go]: removed

File: internal/app/concurrent_proxy_stage_test.go (hunk 1 above)
Comment (miketonks-form3) [reply internal/app/concurrent_proxy_stage_test.go]: I intended to change all to assert, mainly for consistence, just overlooked this file.  Will change.

**Pull request merged**